packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -114,14 +114,15 @@ export abstract class OptimisticOracleClient<R extends OptimisticOracleRequest>
   async updateWithBlockRange(blockRange?: [number, number], existingRequests: R[] = []): Promise<this> {
     let range: [number, number];
     if (blockRange) {
+      if (blockRange[0] > blockRange[1]) throw new Error("Invalid block range");
       range = blockRange;
     } else {
       const latestBlock = await this.provider.getBlockNumber();
-      range = [this.fetchedBlockRange[1] + 1, latestBlock];
+      const nextStartBlock = this.fetchedBlockRange[1] + 1;
+      range = [nextStartBlock, latestBlock >= nextStartBlock ? latestBlock : nextStartBlock];
```

Comment:
If latestBlock < nextStartBlock, this gets invalid range of [nextStartBlock, nextStartBlock] and fetchOracleRequests(range) might fail as trying to get events from future blocks not yet mined